### PR TITLE
EDM-622: Fix local registry deploy targets

### PR DESCRIPTION
**Pull request by oglok**
Without the deploy-e2e-extras the local registry won't be deployed on Kind, so agent-vm scripts will fail.

### CODE REVIEW

File: docs/developer/README.md [hunk 1]
```diff
@@ -64,7 +64,7 @@ this only works on a Linux host.
 ```
 # will create the cluster, and the agent config files in bin/agent which will be embedded in the image
 make deploy
-make agent-vm agent-vm-console # user/password is user/user
+make deploy-e2e-extras agent-vm agent-vm-console # user/password is user/user
```
Comment (hexfusion) [inline docs/developer/README.md:67]: thanks for doing this PR I just wonder if the registry should just be part of deploy. as a dev this is an extra step we need to do fairly often.

Comment (oglok) [reply docs/developer/README.md:67]: I guess that makes sense. wdyt @mangelajo ?

Comment (mangelajo) [reply docs/developer/README.md:67]: make it a dependency to deploy? that would make sense I think, I don't remember now if there was any technical reason why I didn't add it to deploy. I know I split it from the deploy script to be able to do it to a target cluster instead (OpenShift),

https://github.com/flightctl/flightctl/blob/main/deploy/deploy.mk#L9

We could add it here right before the prepare-agent

Comment (oglok) [reply docs/developer/README.md]: Updated!

**Review by mangelajo — APPROVED** [review]

**Pull request merged**